app/services/collect_inventories_service.rb

```diff
@@ -0,0 +1,46 @@
+class CollectInventoriesService
+  attr_reader :inventory_ids, :inventory_tags
+
+  def initialize(options)
+    @options = options.deep_symbolize_keys
+    @service_offering_id = @options[:service_offering_id]
+    @task_id = @options[:task_id]
+    @inventory_ids = []
+    @inventory_tags = []
+  end
+
+  def process
+    Task.update(@task_id, :state => "running", :status => "ok")
+
+    visited = []
+    collect_inventory(@service_offering_id, visited, @inventory_ids)
```

Comment:
@hsong-rh the inventory_id can be a local variable now it is not needed by the caller anymore since they would be going after tags